### DISCUSSION TITLE
Simplest way to deactivate the tqdm progress bar?

### DISCUSSION THREAD

**Discussion by pbruneau** [awslabs/gluonts#1430] [Q&A] [answered]
I have been looking for a simple way to deactivate the tqdm progress bar when training estimators (for example in the context of a batch script). I found no related discussion (no results when searching for "tqdm" or "progress"), and the closest related issue seems to be #810 - but the latter did not translate into apparent API parametrization as far as I could see.

The _simplest_ way I found is to fork gluonts, and modify L29 in env.py with these lines:
```
    tqdm_env = os.environ.get(
        "USE_TQDM", '1'
    )
    try:
        tqdm_env = int(tqdm_env)
    except ValueError:
        tqdm_env = 1

    use_tqdm: bool = bool(tqdm_env)
```
One then has to set ``os.environ['USE_TQDM']`` to '0' or '1' before importing gluonts. Does anyone know if I missed some obvious way?

**Comment by jaheba** [comment, marked as answer]
Hey, you should be able to disable tqdm by using:

```py
from gluonts.env import env

env._push(use_tqdm=False)
```

That disables it here:

https://github.com/awslabs/gluon-ts/blob/ecb903f5a4da4fb0d26f27c61b5e5f8a2d6821b1/src/gluonts/gluonts_tqdm.py#L27


Is that what you need?

**Reply by pbruneau** [reply]
Yes, this is exactly what I need! Way simpler than my hack :)

Thanks for the howto!

**Reply by jaheba** [reply]
Great! Documentation is surely something we need to improve on in the future.
